web/hosts_next.go

```diff
@@ -34,12 +50,40 @@ func NewHostListNextHandler(hostsService services.HostsNextService) gin.HandlerF
 		pagination := NewPaginationWithStrings(len(hostList), page, perPage)
 		firstElem, lastElem := pagination.GetSliceNumbers()
 
+		hContainer := NewHostsNextHealthContainer(hostList)
+		hContainer.Layout = "horizontal"
+
 		c.HTML(http.StatusOK, "hosts_next.html.tmpl", gin.H{
-			"Hosts":          hostList[firstElem:lastElem],
-			"AppliedFilters": query,
-			"FilterSIDs":     filterSIDs,
-			"FilterTags":     filterTags,
-			"Pagination":     pagination,
+			"Hosts":           hostList[firstElem:lastElem],
+			"AppliedFilters":  query,
+			"FilterSIDs":      filterSIDs,
+			"FilterTags":      filterTags,
+			"Pagination":      pagination,
+			"HealthContainer": hContainer,
 		})
 	}
 }
+
+type SendHeartBeat struct {
+	AgentID string `json:"agent_id"`
+}
+
+func ApiHostHeartbeatHandler(hostService services.HostsNextService) gin.HandlerFunc {
+	return func(c *gin.Context) {
+		var sendHeartBeat SendHeartBeat
+
+		err := c.BindJSON(&sendHeartBeat)
+		if err != nil {
+			_ = c.Error(BadRequestError("problems parsing JSON"))
+			return
+		}
+
+		err = hostService.Heartbeat(sendHeartBeat.AgentID)
```

Comment:
This would add a lot of complexity. I think we trust by the fact that the API endpoint is authenticated that we have no byzantine agents, and we could also prune heartbeat from agents that we don't hear from a X period.

Reply:
All you say is correct and I agree.
I understand the complexity that would add, indeed I was not proposing to change/add anything :smile: 

Sorry if I haven't been clear enough. (btw, question: was it clear that I was not proposing to change anything? I thought so, but if it wasn't I'd need to improve)

Reply:
Yes it was, fact is I don't see this in the near future (as long as the users don't come to us asking for it )

Reply:
Yup, absolutely makes sense :wink: